pra-07/src/views/Home.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <div class="container">
+    <div>
+      <AlertBox
+        v-if="this.isLoading == false && this.successmsg !== ''"
+        variant="success"
+        show="true"
+        :content="this.successmsg"
+      />
+      <AlertBox
+        v-else-if="this.isLoading == false && this.errmsg !== ''"
+        variant="danger"
+        show="true"
+        :content="this.errmsg"
+      />
+    </div>
+    <b-row>
+      <GalleryCard
+        v-for="car in this.$store.state.cars.cars"
+        :key="car.id"
+        :cardId="car.id"
+        :carTitle="car.name"
+        :carImage="car.image"
+        :cardetails="car.details"
+        :price="car.price"
+        @CarInfo="CarInfo"
+        @editCard="editCard(car)"
+        @deleteCard="deleteCard(car)"
+      />
+      <transition name="bounce">
+        <FormModal
+          :formData="selectedCardData"
+          @submittedFormData="handleSubmittedData"
+        />
+      </transition>
+    </b-row>
+  </div>
+</template>
+
+<script>
+import GalleryCard from "../components/GalleryCard.vue";
+import FormModal from "../components/Form-Modal.vue";
+import AlertBox from "../components/AlertBox.vue";
+export default {
+  name: "HomeComponent",
+  components: {
+    GalleryCard,
+    FormModal,
+    AlertBox,
+  },
+  props: ["modalId"],
+  data() {
+    return {
+      cars: [],
+      formModalId: this.modalId,
+      selectedCardData: {
+        carId: "",
+        carName: "",
+        carDetails: "",
+        carPrice: "",
+        carImgURL: "",
+      },
+      submittedNames: [],
+      isLoading: false,
+      errmsg: "",
+      successmsg: "",
+    };
+  },
+
+  methods: {
+    CarInfo(id) {
+      this.$router.push({
+        name: "cardetails",
+        params: {
+          carId: id,
+        },
+      });
+    },
+    editCard(car) {
+      this.selectedCardData = {
+        carId: car.id,
+        carName: car.name,
+        carDetails: car.details,
+        carPrice: car.price,
+        carImgURL: car.image,
+      };
+      this.$bvModal.show("modal-prevent-closing");
+    },
+    async deleteCard(data) {
+      let res = await this.$store.dispatch("cars/deleteCarData", data.id);
+      if (res === "success") {
+        this.successmsg = "car data delete successfully";
+      } else {
+        this.errmsg = "oops! somthing went wrong";
```

Comment:
in this.errormsg you can directly assign res because you have return error message in your deleteCarData action.